2.7.0 breaks existing typescript code

this code:


```
//file1.ts
import gql from 'graphql-tag';

export const someMutation = gql`someMutation`;
```
```
//file2.ts
import { ApolloClient } from 'apollo-client';
import {someMutation} from './file1.ts';

const apollo = new ApolloClient(...);
apollo.mutate({
      mutation: someMutation ,
      variables: {
//..
      }
    })
```


Those are the dependencies that work
```
@types\node@8.5.10
@types\node-fetch@1.6.7
apollo-cache-inmemory@1.1.7
apollo-client@2.2.2
apollo-link@1.0.7
apollo-link-context@1.0.3
apollo-link-http@1.3.2
graphql@0.11.7
graphql-tag@2.6.1
node-fetch@1.7.3
ts-node@4.1.0
typescript@2.7.1
```

but when updating to graphql-tag@2.7.0 we get and error:

```
error TS4023: Exported variable 'someMutation ' has or is using name 'DocumentNode' from external module "<my-project>node_modules/@types/graphql/language/ast" but cannot be named.
src/index.ts(38,3): error TS4053: Return type of public method from exported class has or is using name 'ExecutionResult' from external module "<my-project>/node_modules/@types/graphql/execution/execute" but cannot be named.
```

staying with 2.6.1 for now. Thanks!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,10 @@
 # Change log
 
+### vNext
+
+* Update/fix the existing TS `index.d.ts` declaration file.  <br/>
+  [@hwillson](https://github.com/hwillson) in [#285](https://github.com/apollographql/graphql-tag/pull/285)
+
 ### v2.10.1
 
 * Fix failures in IE11 by avoiding unsupported (by IE11) constructor arguments to `Set` by [rocwang](https://github.com/rocwang) in [#190](https://github.com/apollographql/graphql-tag/pull/190)
@@ -7,7 +12,7 @@
 ### v2.10.0
 * Add support for `graphql@14` by [timsuchanek](https://github.com/timsuchanek) in [#210](https://github.com/apollographql/graphql-tag/pull/210), [#211](https://github.com/apollographql/graphql-tag/pull/211)
 
-### v2.9.1 
+### v2.9.1
 * Fix IE11 support by using a regular for-loop by [vitorbal](https://github.com/vitorbal) in [#176](https://github.com/apollographql/graphql-tag/pull/176)
 
 ### v2.9.0

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,3 +1,20 @@
-export default function gql(literals: any, ...placeholders: any[]): any;
-export function resetCaches(): void;
-export function disableFragmentWarnings(): void;
+declare module "graphql-tag" {
+  function gql(
+    literals: ReadonlyArray<string>,
+    ...placeholders: any[]
+  ): import("graphql").DocumentNode
+
+  namespace gql {
+    function resetCaches(): void;
+    function disableFragmentWarnings(): void;
+    function enableExperimentalFragmentVariables(): void;
+    function disableExperimentalFragmentVariables(): void;
+  }
+
+  export default gql;
+
+  export function resetCaches(): void;
+  export function disableFragmentWarnings(): void;
+  export function enableExperimentalFragmentVariables(): void;
+  export function disableExperimentalFragmentVariables(): void;
+}

